code-challenges.test.js

```diff
@@ -19,33 +27,239 @@ const people = [
+
+    expect(jobPlusName(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// ReferenceError: jobPlusName is not defined, good fail
+
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 // b) Create the function that makes the test pass.
 
+// PSEUDOCODE:
+// create function named jobPlusName
+  // takes in an array (of objects)
+// REMEMBER to use array.name or array.occupation to access those values
+// create a container called newArray = []
+// iterate through the array of objects
+  // for each name, separate the first and last name
+    // capitalize the FIRST LETTER of both the first and last name 
+      // concatenate them together again
+  // occupation just needs to be concatenated with + "."
+  // .push() all the pieces to make a sentence
+    // example: "Ford Prefect is a hitchhiker." etc etc
+      // use the container (newArray)
+// return the container
+
+const jobPlusName = (array) => {
+  let newArray = []; // <---- container for final output
+  for (let i = 0; i < array.length; i++) { // <---- iteration through array of objects
+    let firstName = array[i].name.split(" ")[0]; // <---- First name using .split(" ")
+    let lastName = array[i].name.split(" ")[1]; // <---- same thing, but with Last name
+    let firstLetter = firstName.charAt(0).toUpperCase(); // <---- Capitalizing only the first letter of the FIRST name
+    let lastLetter = lastName.charAt(0).toUpperCase(); // <---- Capitalizing only the first letter of the LAST name
+    let newName = firstLetter + firstName.slice(1) + " " + lastLetter + lastName.slice(1); // <--- putting it all back together to make a capitalized name!
+    newArray.push(newName + " is " + array[i].occupation + "."); // <---- finally using the container and creating a sentence, all at once
+  }
+  return newArray; // <---- returning the container's values
+}
+
+console.log(jobPlusName(people))
+
+// Result:
+// console.log
+//     ford
+//       at jobPlusName (code-challenges.test.js:68:13)
+
+//     prefect
+//       at jobPlusName (code-challenges.test.js:70:13)
+
+//     F
+//       at jobPlusName (code-challenges.test.js:72:13)
+
+//     P
+//       at jobPlusName (code-challenges.test.js:74:13)
+
+//     Ford Prefect
+//       at jobPlusName (code-challenges.test.js:76:13)
+      
+// (It iterates through the rest of the array...)
+
+//       [
+//         'Ford Prefect is a hitchhiker.',
+//         'Zaphod Beeblebrox is president of the galaxy.',
+//         'Arthur Dent is a radio employee.'
+//       ]
+
+// WHAT I LEARNED: Console.log() is your friend. I used it on almost every line (I deleted them all to make it more readable) so I could track exactly what was happening and where. This problem took me the longest to figure out so, I left it for last. 
+
```

Comment:
It certainly does help. I know it helps to see what the variables are producing. Great observation